hbase-server/src/main/java/org/apache/hadoop/hbase/HDFSBlocksDistribution.java

```diff
@@ -122,14 +134,27 @@ public synchronized String toString() {
    * @param weight the weight
    */
   public void addHostsAndBlockWeight(String[] hosts, long weight) {
+    addHostsAndBlockWeight(hosts, weight, null);
+  }
+  /**
+   * add some weight to a list of hosts, update the value of unique block weight
+   * @param hosts the list of the host
+   * @param weight the weight
+   */
+  public void addHostsAndBlockWeight(String[] hosts, long weight, StorageType[] storageTypes) {
     if (hosts == null || hosts.length == 0) {
       // erroneous data
       return;
     }
 
     addUniqueWeight(weight);
-    for (String hostname : hosts) {
-      addHostAndBlockWeight(hostname, weight);
+    for (int i = 0; i < hosts.length; i++) {
+      long weightForSsd = 0;
+      if (storageTypes != null && storageTypes.length == hosts.length
+        && storageTypes[i] == StorageType.SSD) {
+        weightForSsd = weight;
```

Comment:
Can we break this if condition? How about this?
```
if (storageTypes != null && storageTypes.length == hosts.length){
  for (int i = 0; i < hosts.length; i++) {
    long weightForSsd = 0;
    if (storageTypes[i] == StorageType.SSD){
      weightForSsd = weight;
    }
    addHostAndBlockWeight(hosts[i], weight, weightForSsd);
  }
}
```

Reply:
It seems a bit different, since we should add the weight no matter what storageTypes is.

Reply:
Thanks @bsglz , I got your point. So, maybe this one is better?
```
if (storageTypes != null && storageTypes.length == hosts.length){
  for (int i = 0; i < hosts.length; i++) {
    long weightForSsd = 0;
    if (storageTypes[i] == StorageType.SSD){
      weightForSsd = weight;
    }
    addHostAndBlockWeight(hosts[i], weight, weightForSsd);
  }
} else {
  for (String hostname : hosts) {
    addHostAndBlockWeight(hostname, weight, 0);
  }
}
```
The point is for each host iteration, checking `storageTypes != null && storageTypes.length == hosts.length` seems redundant. What do you think?

Reply:
Yeah, it seems better, thanks.